repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -197,18 +230,62 @@ object Evaluator{
       // Exhaust the printer iterator now, before exiting the `Catching`
       // block, so any exceptions thrown get properly caught and handled
       evaluatorRunPrinter(printer(evalMain(cls).asInstanceOf[Iterator[String]]))
+      evaluationResult(wrapperName, newImports)
+    }
+
+    def compileCacheBlock(wrapperName: String, code: String, scriptImports: Seq[ImportData]): Res[(Class[_], Seq[ImportData])] = for {
```

Comment:
Yeah let's do that. compile-caching predef would be great, since it will rarely change, and people would pay the cost at every initialization!